### PR TITLE
Patch FFTW detection in libvdwxc

### PR DESCRIPTION
**Pull request by lang-m**
With GCC 14.1 configure fails during the FFTW detection with:
```
conftest.c:42:26: error: passing argument 1 of 'fftw_execute_dft' from incompatible pointer type [-Wincompatible-pointer-types]
   42 |         fftw_execute_dft(plan, a1, a2);
      |                          ^~~~
      |                          |
      |                          struct fftw_plan_s **
```

The same fix would also be required for `config/m4/fftw3.m4`. I did not change it because `configure` is not re-created during the `autoreconf` step and this seems in line with https://spack.readthedocs.io/en/latest/build_systems/autotoolspackage.html#patching-configure-or-makefile-in-files. If require, I can update the patch.

### CODE REVIEW

File: var/spack/repos/builtin/packages/libvdwxc/package.py
Comment (lang-m) [inline var/spack/repos/builtin/packages/libvdwxc/package.py]: The restriction on `gcc14:` is probably not required. Should I remove it?

Comment (bernhardkaindl) [reply var/spack/repos/builtin/packages/libvdwxc/package.py]: @lang-m: Yes, in principle, passing the correct variable type should not hurt in configure, it may help with other compilers too after all.

Normally, I'd say that it's safer to not change the configure for other compilers but in this case, as far as I can see, the configure test was using a pointer to the type when it should be just the type.

Because the `configure` is generated, one should possibly fix the actual source as well and get the fix for the actual source file merged upstream.

This would also help if one enables regenerating configure in the build in `spack. (which is currently apparently not needed),

This would be the source location of the source fix if regenerating configure would be added to the spac package, and the fix to upstream:
https://gitlab.com/libvdwxc/libvdwxc/-/blob/master/config/m4/fftw3.m4?ref_type=heads#L54

Just BTW: After 4 years of absence, I saw that https://gitlab.com/askhl merged a few pull requests.

Could you also raise this with https://gitlab.com/askhl upstream?
https://gitlab.com/libvdwxc/libvdwxc/-/blob/master/config/m4/fftw3.m4?ref_type=heads#L54 

Then, a future release should include the fix.

**Review by bernhardkaindl — DISMISSED** [review]
@lang-m Approving nonetheless, but it would be great if you could contact https://gitlab.com/askhl to merge the actual source fix upstream.

Holding off the merge because of these possible improvements and the upstream submission.

Please ping me or some other maintainer if the PR is fully finished for merge.

**Comment by lang-m** [comment]
Upstream MR: https://gitlab.com/libvdwxc/libvdwxc/-/merge_requests/13

I will update this PR once receiving feedback upstream.

File: var/spack/repos/builtin/packages/libvdwxc/package.py
Comment (lang-m) [reply var/spack/repos/builtin/packages/libvdwxc/package.py]: The fix has been merged upstream (https://gitlab.com/libvdwxc/libvdwxc/-/merge_requests/13) without further modification. I have updated the package.py file to also use the upstream fix for the m4 macro in case `configure` would be re-generated.

I have removed the restriction on the compiler and instead limited the patches to currently released versions of libvdwxc, which come without the fix.

**Comment by lang-m** [comment]
@bernhardkaindl From my side this is ready to be merged.

**Comment by bernhardkaindl** [comment]
@spackbot run pipeline

**Review by bernhardkaindl — APPROVED** [review]
Thanks for the upstream merge, perfect!

**Pull request merged**
